### DISCUSSION TITLE
Middleware is on edge runtime and can't be changed to nodejs. How to work around?

### DISCUSSION THREAD

**Discussion by paulwongx** [vercel/next.js#51836] [App Router]
I'm using middleware to validate the user session. This requires `import { Client } from "pg";` but the pg package requires nodejs packages like the `net` module.

How can I get the middleware to work even when requiring nodejs modules?

My thoughts:
1) Do I need to have another route using the default nodejs runtime and redirect to that route from the middleware and then call NextResponse.next() or something?
2) Alternatively is there a postgres client that can connect on the edge and works well with drizzle?
3) Or is there some way to change middleware to run on nodejs?

**Comment by icyJoseph** [comment]
Uff not sure, I think the official one is, https://vercel.com/docs/storage/vercel-postgres/sdk, but that might not be what you need. 

Right now, maybe you'd need to redirect the user to an identify endpoint, which the middleware should ignore, include the path they intended to use in the URL as a query, and once identified, redirect them back to their original intention. Default that to `/` if not present.
